### PR TITLE
Solid surface colour

### PR DESCRIPTION
**Pull request by neworderofjamie**
For prototyping virtual environments, using solid colours rather than properly texturing everything saves loads of time. This PR implements support for this when loading obj files into antworld. Additionally it fixes a couple of windows build issues and improves the CLI to the ant world example.

### CODE REVIEW

**Comment by neworderofjamie** [comment]
@alexdewar any chance you could have a quick look at this? I'm onto Stefan's _next_ request and I'd rather not end up in too much of a branch mess...

**Review by alexdewar — COMMENTED** [review]
I added a couple of minor comments before... It looks good to me except I just wanted a bit of clarification with the ``m_Textures`` thing.

File: src/antworld/world.cc [hunk 1]
```diff
@@ -513,11 +536,30 @@ void World::loadMaterials(const filesystem::path &basePath, const std::string &f
...
+                    // Set texture in material
+                    std::get<0>(mtl->second) = m_Textures.back().get();
+                }
             }
         }
+        // Otherwise, if it's another material property, ignore
+        else if(commandString == "Ns" || commandString == "Ka" || commandString == "Ks" ||
+                commandString == "Ke" || commandString == "Ni" || commandString == "d" ||
+                commandString == "illum")
+        {
```
[the remark this answers is not in the document]
Comment (neworderofjamie) [reply src/antworld/world.cc:560]: all mtl files will have these (they specify more complex material parameters to do with lighting) so this is us knowing what they are but ignoring them

[the remark this answers is not in the document]
Comment (alexdewar) [reply src/antworld/world.cc:560]: Fair dos!

File: src/antworld/world.cc [hunk 2]
```diff
@@ -513,11 +536,30 @@ void World::loadMaterials(const filesystem::path &basePath, const std::string &f
                 m_Textures.emplace_back(new Texture());
                 m_Textures.back()->upload(texture, textureFormat);
```
[the remark this answers is not in the document]
Comment (neworderofjamie) [reply src/antworld/world.cc:537]: ``m_Textures`` is only deleted when the object's deleted (which isn't going to happen mid-way through ``loadObj``) whereas ``materialNames`` is only used within the scope of ``loadObj``. However, ``m_Textures`` should make it's ownership automatic and explicit by using ``std::unique_ptr`` - will change now

[the remark this answers is not in the document]
Comment (neworderofjamie) [reply src/antworld/world.cc:537]: Actually, ``m_Textures`` already uses ``std::unique_ptr`` so I _think_ this is fine?

[the remark this answers is not in the document]
Comment (alexdewar) [reply src/antworld/world.cc:537]: Yep, that's fine then. I was just a bit worried that the scopes weren't overlapping.

**Pull request merged**